tests/D2L.CodeStyle.Analyzers.Test/SpecRunner.cs

```diff
@@ -49,27 +31,34 @@ private sealed class DiagnosticComparer : IEqualityComparer<Diagnostic> {
 
 			var analyzer = GetAnalyzerNameFromSpec( test.Source );
 
-			var compilation = GetCompilationForSource( test.Name, test.Source, test.MetadataReferences );
+			Compilation compilation = GetCompilationForSource( test.Name, test.Source, test.MetadataReferences );
+			CompilationUnitSyntax compilationUnit = (CompilationUnitSyntax)compilation.SyntaxTrees.First().GetRoot();
 
-			m_actualDiagnostics = GetActualDiagnostics( compilation, analyzer, test.AdditionalFiles );
+			m_actualDiagnostics = GetActualDiagnostics( compilation, analyzer, test.AdditionalFiles )
+				.Select( PrettyDiagnostic.Create )
+				.ToImmutableArray();
 
-			m_expectedDiagnostics = GetExpectedDiagnostics(
-				(CompilationUnitSyntax)compilation.SyntaxTrees.First().GetRoot(),
-				test.DiagnosticDescriptors
-			).ToImmutableArray();
+			m_expectedDiagnostics = GetExpectedDiagnostics( compilationUnit, test.DiagnosticDescriptors )
+				.Select( PrettyDiagnostic.Create )
+				.ToImmutableArray();
 
 			m_matchedDiagnostics = m_actualDiagnostics
-				.Intersect(
-					m_expectedDiagnostics,
-					DiagnosticComparer.Instance
-				).ToImmutableHashSet( DiagnosticComparer.Instance );
+				.Intersect( m_expectedDiagnostics )
+				.ToImmutableHashSet();
 		}
 
 		[Test]
 		public void NoUnexpectedDiagnostics() {
-			var unexpectedDiagnostics = m_actualDiagnostics
+
+			IEnumerable<PrettyDiagnostic> unexpectedDiagnostics = m_actualDiagnostics
 				.Where( d => !m_matchedDiagnostics.Contains( d ) );
 
+			Assert.Multiple( () => {

```

Comment:
Using `Assert.Multiple` to gather all 

![image](https://user-images.githubusercontent.com/12101120/152999755-6ec761f4-9349-4bc8-8212-261dcf764a27.png)